### PR TITLE
[dynamo] Support pytree flattening data structures with KJT.

### PR DESCRIPTION
**Pull request by zhxchen17**
Summary: Register the rule for flattening KJT in dynamo so that we can unblock torch.export() for some important server use cases.

Test Plan: buck2 run mode/opt sigmoid/frontend:unflatten_test -- -r kjt_dynamo_flatten

Differential Revision: D48833738

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov

### CODE REVIEW

**Comment by ezyang** [comment]
See also https://github.com/pytorch/pytorch/issues/107315

To be honest, I have half the mind to make you do it properly by fixing the underlying issue. Change my mind.

**Comment by zhxchen17** [comment]
> See also #107315
> 
> To be honest, I have half the mind to make you do it properly by fixing the underlying issue. Change my mind.

#108567 @ezyang tbh https://github.com/pytorch/pytorch/pull/108533 looks much better than this lol. I will test on that PR instead.

**Comment by zhxchen17** [comment]
in favor of https://github.com/pytorch/pytorch/pull/108533
